### ISSUE TITLE
Failed at "Cypress start:mocks" during CI workflow

### ISSUE THREAD

**Issue by test-2287** [kentcdodds/onewheel-blog#7]
Hi, i followed the [egghead course](https://egghead.io/lessons/remix-deploy-a-remix-application-to-fly-io) and the README file to deploy, but got `throw new Error(prefix);` at `tiny-invariant`

<img width="791" alt="image" src="https://user-images.githubusercontent.com/27443988/192039639-d4ad9156-0ea1-4b35-b16c-1bdd69eb0217.png">

and i found `npm run start:mocks` have the same issue in my local dev env.

node version:  v16.16.0
npm version: 8.11.0
platform: macOS bigSur

**Comment by kentcdodds** [comment]
Did you run the `fly secrets set` command before deploying?

**Comment by test-2287** [comment]
@kentcdodds yes~ , i ran `fly secrets set SESSION_SECRET=$(openssl rand -hex 32) --app blog-tutorial-c08c` showed in the README and ran `fly secrets set ADMIN_EMAIL=rachel@remix.run` show in the course.

<img width="564" alt="image" src="https://user-images.githubusercontent.com/27443988/192082293-3056afe8-6cf8-421f-9a02-f3e591e76ad6.png">

**Comment by kentcdodds** [comment]
Yeah, that's quite strange... I'm sorry, I don't know what it could be 😬

**Comment by test-2287** [comment]
> Yeah, that's quite strange... I'm sorry, I don't know what it could be 😬

i found cypress ran start:mocks failed cause it can't get the env variables, since the default NODE_ENV of remix serve run is `production`, it will always throw Error in tiny-invariant.js 

```
var isProduction = process.env.NODE_ENV === 'production';
var prefix = 'Invariant failed';
function invariant(condition, message) {
    if (condition) {
        return;
    }
    if (isProduction) {
        throw new Error(prefix);
    }
    var provided = typeof message === 'function' ? message() : message;
    var value = provided ? prefix + ": " + provided : prefix;
    throw new Error(value);
}

module.exports = invariant;
```

i added the env as below in deploy.yml file and it works, i not sure if there is another better solution.
<img width="468" alt="image" src="https://user-images.githubusercontent.com/27443988/192135864-a7dc1c3c-c451-4397-954d-36a72b029a9d.png">

**Comment by test-2287** [comment]
> Yeah, that's quite strange... I'm sorry, I don't know what it could be 😬

i found cypress ran start:mocks failed cause it can't get the env variables, since the default NODE_ENV of remix serve run is `production`, it will always throw Error in tiny-invariant.js 

```
var isProduction = process.env.NODE_ENV === 'production';
var prefix = 'Invariant failed';
function invariant(condition, message) {
    if (condition) {
        return;
    }
    if (isProduction) {
        throw new Error(prefix);
    }
    var provided = typeof message === 'function' ? message() : message;
    var value = provided ? prefix + ": " + provided : prefix;
    throw new Error(value);
}

module.exports = invariant;
```

i added the env as below in deploy.yml file and it works, i not sure if there is another better solution.
<img width="468" alt="image" src="https://user-images.githubusercontent.com/27443988/192135864-a7dc1c3c-c451-4397-954d-36a72b029a9d.png">

**Comment by kentcdodds** [comment]
Oh, did you update the `.env.example`? That's important because the github action will copy that file to `.env` before running the E2E tests.

**Comment by amite** [comment]
> Oh, did you update the `.env.example`? That's important because the github action will copy that file to `.env` before running the E2E tests.

I am getting the same error. I tried updating the .env.example and it still crashes. I'll file a separate issue

**Comment by bbaki** [comment]
For anyone else trying to test the build locally, remix does not use `.env` file by default ([link](https://remix.run/docs/en/1.14.3/guides/envvars#md-local-development)), also see [this issue](https://github.com/remix-run/remix/issues/5341).

The [solution ](https://github.com/remix-run/remix/issues/5341#issuecomment-1412835063) is to use e.g. `dotenv-cli` like this: `"start": "dotenv -e .env.production remix-serve build"`
